packages/plugin-commands-setup/src/setup.win32.ts

```diff
@@ -0,0 +1,67 @@
+import execa from 'execa'
+
+type IEnvironmentValueMatch = { groups: { name: string, type: string, data: string } } & RegExpMatchArray
+
+export async function setupEnvironmentPath (pnpmHomeDir: string): Promise<string> {
+  const pathRegex = /^ {4}(?<name>PATH) {4}(?<type>\w+) {4}(?<data>.*)$/gim
+  const pnpmHomeRegex = /^ {4}(?<name>PNPM_HOME) {4}(?<type>\w+) {4}(?<data>.*)$/gim
+  const regKey = 'HKEY_CURRENT_USER\\Environment'
+
+  const queryResult = await execa('reg', ['query', regKey])
+
+  if (queryResult.failed) {
+    return 'Win32 registry environment values could not be retrieved'
+  }
+
+  const queryOutput = queryResult.stdout
+  const pathValueMatch = [...queryOutput.matchAll(pathRegex)] as IEnvironmentValueMatch[]
+  const homeValueMatch = [...queryOutput.matchAll(pnpmHomeRegex)] as IEnvironmentValueMatch[]
+
+  let commitNeeded = false
+  let homeDir = pnpmHomeDir
+  const logger = []
+
+  if (homeValueMatch.length === 1) {
+    homeDir = homeValueMatch[0].groups.data
+    logger.push(`Currently 'PNPM_HOME' is set to '${homeDir}'`)
+  } else {
+    logger.push(`Setting 'PNPM_HOME' to value '${homeDir}'`)
+    const addResult = await execa('reg', ['add', regKey, '/v', 'PNPM_HOME', '/t', 'REG_EXPAND_SZ', '/d', homeDir, '/f'])
+    if (addResult.failed) {
+      logger.push(`\t${addResult.stderr}`)
+    } else {
+      commitNeeded = true
+      logger.push(`\t${addResult.stdout}`)
+    }
+  }
+
+  if (pathValueMatch.length === 1) {
+    const pathData = pathValueMatch[0].groups.data
+    if (pathData == null || pathData.trim() === '') {
+      logger.push('Current PATH is empty. No changes to this environment variable are applied')
+    } else {
+      const pathDataUpperCase = pathData.toUpperCase()
+      const homeDirUpperCase = `${homeDir.toUpperCase()};`
+      if (pathDataUpperCase.includes(homeDirUpperCase)) {
```

Comment:
The trick can be reduced (import * as path from 'path')

```suggestion
      if (homeDir.toUpperCase().split(path.delimiter).includes(homeDir.toUpperCase())) {
```